source/docs/software/advanced-gradlerio/deploy-git-data.rst

```diff
@@ -0,0 +1,100 @@
+         // File name to write to
+         "branch.txt"
+      ).text = branch // Set the contents of the file to the variable branch
+   }
+
+This registers a `Gradle task <https://docs.gradle.org/current/userguide/tutorial_using_tasks.html>`__ that uses the above Git command, saves the output to a variable, and then writes it to a file. Since it was written to the ``src/main/deploy`` directory, it will be included in the jar file deployed to the robot and accessible in code.
+
+The next step is to make the deploy task depend on the task you created, so that it will automatically run before the code is deployed. This example uses the task name ``writeBranchName`` from the previous example, but it should be replaced with the name of the task in your ``build.gradle``.
+
+.. code-block:: groovy
+
+   tasks.deploy.dependsOn(writeBranchName)
+
+Deploying Commit Hash
+---------------------
+
+Similar to the previous example, ``git rev-parse`` will be used to parse the current commit hash. The Git command used for this is:
+
+.. code-block:: console
+
+   $ git rev-parse --short HEAD
+
+Similar to the previous Git command, ``rev-parse`` is used to find information about the commit at HEAD. However, instead of using ``--abbrev-ref`` to find the branch name associated with that commit, ``--short`` is used to find the 7-character commit hash.
+
+.. note:: If you wish to use the full commit hash instead of the 7-character version, you can leave out the ``--short`` flag.
+
+Next is to create a task in ``build.gradle`` that runs this command and writes the output to a file. This is largely the same as the first example, except the task is named ``writeCommitHash``, the new Git command is used, and it is written to ``commit.txt`` instead of ``branch.txt``.
+
+.. code-block:: groovy
+
+   tasks.register("writeCommitHash") {
+      def stdout = new ByteArrayOutputStream()
+
+      exec {
+        commandLine "git", "rev-parse", "--short", "HEAD"
+        standardOutput = stdout
+      }
+
+      def commitHash = stdout.toString().trim()
+
+      new File(
+         projectDir.toString() + "src/main/deploy",
+         "commit.txt"
+      ).text = commitHash
+   }
+
+   tasks.deploy.dependsOn("writeCommitHash")
+
+Using Deployed Files
+--------------------
+
+In order to access files that were written to the deploy directory in code, you have to use the ``getDeployDirectory()`` method of the `Filesystem <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/Filesystem.html>`__ class in Java, or the ``GetDeployDirectory()`` function of the `frc::filesystem <https://first.wpi.edu/FRC/roborio/release/docs/cpp/namespacefrc_1_1filesystem.html>`__ namespace in C++. Below is an example of opening both files from the previous examples:
+
+.. code-block:: java
+
+   File deployDir = Filesystem.getDeployDirectory();
+   File branchFile = new File(deployDir, "branch.txt");
+   File commitFile = new FIle(deployDir, "commit.txt");
```

Comment:
Perhaps we should mention that these are expensive blocking tasks, and not to do them in any periodic function.

Reply:
Added a note that says this right above